src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1681,6 +1684,27 @@ private function dumpValue($value, $interpolate = true)
         return $this->export($value);
     }
 
+    private function dumpServiceClosure(Reference $reference, $interpolate, $oneLine)
+    {
+        $type = '';
+        if (PHP_VERSION_ID >= 70000 && $reference instanceof AutowirableReference) {
+            $type = $reference->getType();
+            if (ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE === $reference->getInvalidBehavior()) {
+                $type = '\\'.$type;
+            } elseif (PHP_VERSION_ID >= 70100) {
+                $type = '?\\'.$type;
+            } else {
+                $type = '';
+            }
+        }
+
+        if ($oneLine) {
+            return sprintf("function ()%s { return %s; }", $this->dumpValue($reference, $interpolate));
+        }
+
+        return sprintf("function ()%s {\n                return %s;\n            }", $this->dumpValue($reference, $interpolate));
```

Comment:
why? indentation is desired in the output
the type is not mandatory, always putting the `:` would be a parse error